### PR TITLE
Add new IntelliSense modes to support cross compilation configuration

### PR DESCRIPTION
**Pull request by michelleangela**
- Update API version to 5.0.0
- Add 28 new IntelliSense modes

This supports https://github.com/microsoft/vscode-cpptools/pull/6745

### CODE REVIEW

**Review by sean-mcmanus — APPROVED** [review]

File: api.ts [hunk 1]
```diff
@@ -153,9 +153,16 @@ export interface SourceFileConfiguration {
     readonly defines: string[];
 
     /**
-     * The compiler to emulate.
+     * The platform, compiler, and architecture variant to emulate.
      */
-    readonly intelliSenseMode?: "msvc-x86" | "msvc-x64" | "msvc-arm" | "msvc-arm64" |
+    readonly intelliSenseMode?: "linux-clang-x86" | "linux-clang-x64" | "linux-clang-arm" | "linux-clang-arm64" |
```
Comment (sean-mcmanus) [inline api.ts:158]: Should this be formatted better to make it easier to read/compare? Maybe all the platforms on a singe line?

Comment (bobbrow) [reply api.ts:158]: I think it is fine this way.

File: api.ts [hunk 2]
```diff
@@ -153,9 +153,16 @@ export interface SourceFileConfiguration {
     readonly defines: string[];
 
     /**
-     * The compiler to emulate.
+     * The platform, compiler, and architecture variant to emulate.
```
Comment (bobbrow) [inline api.ts:156]: Can we clarify somehow that the enum values without a platform will resolve to the host platform?

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "vscode-cpptools",
-  "version": "4.0.3",
+  "version": "5.0.0",
```
Comment (bobbrow) [inline package.json:3]: We also need to update the `export enum Version` in `api.ts` and add a `v5` (which will be consumed by cpptools)

**Review by bobbrow — CHANGES_REQUESTED** [review]

**Review by bobbrow — APPROVED** [review]

**Pull request merged**